app/services/page_updater.rb

```diff
@@ -0,0 +1,92 @@
+class PageUpdater
```

Comment:
Could you write some rdoc documentation about what this class does and how it is used.